src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -2982,6 +2983,34 @@ err_t sockinfo_tcp::syn_received_drop_lwip_cb(void *arg, struct tcp_pcb *newpcb,
 	return ERR_ABRT;
 }
 
+err_t sockinfo_tcp::dst_nc_send_lwip_cb(void *arg, struct tcp_pcb *newpcb, err_t err)
+{
+	sockinfo_tcp *listen_sock = (sockinfo_tcp *)((arg));
+
+	if (!listen_sock || !newpcb) {
+		return ERR_VAL;
+	}
+
+	sockinfo_tcp *new_sock = (sockinfo_tcp *)((newpcb->my_container));
+
+	NOT_IN_USE(err);
+
+	ASSERT_LOCKED(listen_sock->m_tcp_con_lock);
+	listen_sock->m_tcp_con_lock.unlock();
+
+	new_sock->set_conn_properties_from_pcb();
+	new_sock->create_dst_entry();
+
+	if (new_sock->m_p_connected_dst_entry) {
+		new_sock->prepare_dst_to_send(false);
+		new_sock->abort_connection();
+	}
+	close(new_sock->get_fd());
+
+	listen_sock->m_tcp_con_lock.lock();
+
+	return ERR_OK;
+}
```

Comment:
The logic is very similar to : syn_received_lwip_cb / syn_received_drop_lwip_cb
Can we avoid this code duplication ?